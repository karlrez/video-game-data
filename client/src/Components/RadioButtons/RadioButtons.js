import "./RadioButtons.css";
import Arrow from "../../Assets/up-arrow.png";
import ReactTooltip from "react-tooltip";

function RadioButtons(props) {
  return (
    <div className="radio-buttons__wrapper">
      <table className="radio-buttons__table">
        <tbody>
          <tr>
            <td
              id="first"
              className={
                props.buttonVal === "-YearReleased"
                  ? "radio-buttons__td first radio-button--highlight"
                  : "radio-buttons__td first"
              }
              onClick={(e) => props.onClick("-YearReleased")}
              data-tip="Sort by year"
            >
              Year
            </td>
            <td
              id="second"
              className={
                props.buttonVal === "-US Sales (millions)"
                  ? "radio-buttons__td second radio-button--highlight"
                  : "radio-buttons__td second"
              }
              onClick={(e) => props.onClick("-US Sales (millions)")}
              data-tip="Sort by sales"
            >
              Sales
            </td>
            <td
              id="third"
              className={
                props.buttonVal === "-Review Score"
                  ? "radio-buttons__td third radio-button--highlight"
                  : "radio-buttons__td third"
              }
              onClick={(e) => props.onClick("-Review Score")}
              data-tip="Sort by review score"
            >
              Review Score
            </td>
            <td
              id="arrowBtn"
              className={
                props.arrowBtn
                  ? "radio-button__arrow"
                  : "radio-button__arrow arrowDown"
              }
              data-tip={
                props.arrowBtn
                  ? "Change to ascending order"
                  : "Change to descending order"
              }
            >
              <button className="radio-button__arrow__button">
                <img
                  src={Arrow}
                  alt="up-arrow"
                  onClick={props.onArrowClick}
                  id="arrowBtnImage"
                />
              </button>
            </td>
          </tr>
        </tbody>
      </table>
      <ReactTooltip place="top" type="dark" effect="float" />
    </div>
  );
}

export default RadioButtons;
