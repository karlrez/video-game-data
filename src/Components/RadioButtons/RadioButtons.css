.radio-buttons__wrapper {
  width: 100%;
  text-align: center;
  margin-bottom: 2px;
}

.radio-buttons__table {
  margin-left: auto;
  margin-right: auto;
  width: 96%;
}

.radio-buttons__td {
  width: 30%;
  height: 20px;
  padding: 8px;
  color: black;
  cursor: pointer;
  border-radius: 10px;
  background-color: #f8ebce;
  border-style: solid;
}

.radio-button--highlight {
  color: black;
  box-shadow: 0px 0px 5px red;
}

.radio-button__arrow {
  height: 16px;
  width: auto;
  cursor: pointer;
  -moz-transition: transform 1s;
  -webkit-transition: transform 1s;
  transition: transform 1s;
}

.arrowDown {
  transform: rotate(180deg);
  -moz-transform: rotate(180deg);
  -webkit-transform: rotate(180deg);
}

.radio-button__arrow__button {
  background: none;
  outline: none;
  border: none;
  cursor: pointer;
}
