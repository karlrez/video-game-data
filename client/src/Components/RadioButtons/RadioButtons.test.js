import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import RadioButtons from "./RadioButtons";

configure({ adapter: new Adapter() });

let wrapper;
beforeEach(() => {
  wrapper = shallow(<RadioButtons />);
});

afterEach(() => {
  wrapper.unmount();
});

describe("RadioButtons component", () => {
  it("Matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Year btn has correct text", () => {
    expect(wrapper.find("#first").text()).toEqual("Year");
  });

  it("Sales btn has correct text", () => {
    expect(wrapper.find("#second").text()).toEqual("Sales");
  });

  it("Review score btn has correct text", () => {
    expect(wrapper.find("#third").text()).toEqual("Review Score");
  });
});
