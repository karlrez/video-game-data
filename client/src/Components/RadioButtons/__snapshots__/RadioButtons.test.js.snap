// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtons component Matches snapshot 1`] = `ShallowWrapper {}`;
